From 732f80bdb6da2084e2a83e7eab032dc899307ad7 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyich@gmail.com>
Date: Tue, 13 Dec 2022 19:01:29 +0000
Subject: [PATCH] Add missing <cstdint headers for uint*_t types

Without the change buld fails on upcomig `gcc-13` as:

    frmts/mrf/LERCV1/Lerc1Image.cpp: In member function 'bool Lerc1NS::Lerc1Image::isallsameval(int, int, int, int) const':
    frmts/mrf/LERCV1/Lerc1Image.cpp:730:5: error: 'uint32_t' was not declared in this scope
      730 |     uint32_t val = *reinterpret_cast<const uint32_t *>(&(*this)(r0, c0));
          |     ^~~~~~~~
    /build/gdal/frmts/mrf/LERCV1/Lerc1Image.cpp:29:1: note: 'uint32_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
       28 | #include <algorithm>
      +++ |+#include <cstdint>
---
 frmts/mrf/LERCV1/Lerc1Image.cpp                    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/frmts/mrf/LERCV1/Lerc1Image.cpp b/frmts/mrf/LERCV1/Lerc1Image.cpp
index 8859db6da98f..7f72ad8a3513 100644
--- a/frmts/mrf/LERCV1/Lerc1Image.cpp
+++ b/frmts/mrf/LERCV1/Lerc1Image.cpp
@@ -22,6 +22,7 @@ Contributors:  Thomas Maurer
 */
 
 #include "Lerc1Image.h"
+#include <cstdint>
 #include <cmath>
 #include <cfloat>
 #include <string>
 
